pallets/file-system/src/tests.rs

```diff
@@ -579,48 +945,294 @@ fn bsp_stop_storing_success() {
+        assert_eq!(
+            FileSystem::storage_requests(location.clone()),
+            Some(StorageRequestMetadata {
+                requested_at: 1,
+                owner: owner_account_id.clone(),
+                fingerprint,
+                size,
+                user_multiaddresses: Default::default(),
+                data_server_sps: BoundedVec::default(),
+                bsps_required: current_bsps_required.checked_add(1).unwrap(),
+                bsps_confirmed: 0,
+                bsps_volunteered: 0,
+            })
+        );
+
+        // Assert that the correct event was deposited
+        System::assert_last_event(
+            Event::BspStoppedStoring {
+                bsp: bsp_account_id,
+                file_key,
+                owner: owner_account_id,
+                location,
+            }
+            .into(),
+        );
+    });
+}
+
+#[test]
+fn bsp_stop_storing_no_storage_request_success() {
+    new_test_ext().execute_with(|| {
+        let bsp_account_id = AccountId32::new([2; 32]);
+        let bsp_signed = RuntimeOrigin::signed(bsp_account_id.clone());
+        let file_key = H256::from_slice(&[1; 32]);
+        let location = FileLocation::<Test>::try_from(b"test".to_vec()).unwrap();
+        let owner_account_id = AccountId32::new([1; 32]);
+        let size = 4;
+        let fingerprint = H256::zero();
+
+        // Dispatch BSP stop storing.
+        assert_ok!(FileSystem::bsp_stop_storing(
+            bsp_signed.clone(),
+            file_key,
+            location.clone(),
+            owner_account_id.clone(),
+            fingerprint,
+            size,
+            false
+        ));
+
+        // Assert that the storage request was created with one bsps_required
+        assert_eq!(
+            FileSystem::storage_requests(location.clone()),
+            Some(StorageRequestMetadata {
+                requested_at: 1,
+                owner: owner_account_id.clone(),
+                fingerprint,
+                size,
+                user_multiaddresses: Default::default(),
+                data_server_sps: BoundedVec::default(),
+                bsps_required: 1,
+                bsps_confirmed: 0,
+                bsps_volunteered: 0,
+            })
+        );
+
+        // Assert that the correct event was deposited
+        System::assert_last_event(
+            Event::BspStoppedStoring {
+                bsp: bsp_account_id,
+                file_key,
+                owner: owner_account_id,
+                location,
+            }
+            .into(),
+        );
+    });
+}
+
+#[test]
+fn compute_asymptotic_threshold_point_success() {
+    new_test_ext().execute_with(|| {
+        // Test the computation of the asymptotic threshold
+        let threshold = FileSystem::compute_asymptotic_threshold_point(1)
+            .expect("Threshold should be computable");
+
+        // Assert that the computed threshold is as expected
+        assert!(
```

Comment:
This could be more thorough in the check, and actually check the result for a known input of numbers.

Reply:
I don't want to compute the threshold in the unit test because I feel like it is redundant. 
I added a an extra condition though that checks that the threshold should be greater or equal than the asymptote.
`threshold >= <Test as Config>::AssignmentThresholdAsymptote::get(),`